### PR TITLE
Migrate AWSMetrics Gem to CDKv2

### PR DESCRIPTION
**Pull request by lmbr-pip**
Signed-off-by: Pip Potter <61438964+lmbr-pip@users.noreply.github.com>

## What does this PR do?

* Updates AWS Metrics to CDKv2
    * Bumped Lambdas to Py 3.9
    * Archived CDKv1 version
* Not: Gathering all the CDKv2 changes in feature branch (will include the 4 gems plus testing changes).

## How was this PR tested?

```
cdk synth
cdk deploy
```

### CODE REVIEW

**Review by junbo75 — APPROVED** [review]
Looks good to me. Thanks for the migration.

File: Gems/AWSMetrics/cdk/requirements.txt [hunk 1]
```diff
@@ -1,11 +1,2 @@
...
-
-aws-cdk.aws_apigateway>=1.91.0
-aws-cdk.aws_cloudwatch>=1.91.0
-aws-cdk.aws_iam>=1.91.0
-aws-cdk.aws_kinesis>=1.91.0
-aws-cdk.aws_kinesisanalytics>=1.91.0
-aws-cdk.aws_kinesisfirehose>=1.91.0
-aws-cdk.aws_s3>=1.91.0
-aws-cdk.aws_glue>=1.91.0
-aws-cdk.aws_athena>=1.91.0
```
Comment (allisaurus) [inline Gems/AWSMetrics/cdk/requirements.txt:11]: don't we still need the per-service libs? or are these folded into `aws-cdk-lib` now?

Comment (junbo75) [reply Gems/AWSMetrics/cdk/requirements.txt:11]: I think these are all included in aws-cdk-lib in CDK v2.

Comment (lmbr-pip) [reply Gems/AWSMetrics/cdk/requirements.txt:11]: Yep they are in vended from the main cdk lib and not separated.

**Review by allisaurus — APPROVED** [review]

**Pull request merged**